From 4ae472317fcd0cbcec80ed444de8300bac7cfd6f Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 27 Feb 2013 18:35:09 +0200
Subject: [PATCH 104/177] bone-capemgr: Fix crash when trying to remove
 non-existant slot

Wrong test for not-found case lead to crash. Fix.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 drivers/misc/cape/beaglebone/capemgr.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/misc/cape/beaglebone/capemgr.c b/drivers/misc/cape/beaglebone/capemgr.c
index 2656e3a..9f8da94 100644
--- a/drivers/misc/cape/beaglebone/capemgr.c
+++ b/drivers/misc/cape/beaglebone/capemgr.c
@@ -990,15 +990,12 @@ static ssize_t slots_store(struct device *dev, struct device_attribute *attr,
 		mutex_lock(&info->slots_list_mutex);
 		list_for_each_entry(slot, &info->slot_list, node) {
 			if (slotno == slot->slotno)
-				break;
-		}
-
-		/* found? */
-		if (slot == NULL) {
-			mutex_unlock(&info->slots_list_mutex);
-			return -ENODEV;
+				goto found;
 		}
 
+		mutex_unlock(&info->slots_list_mutex);
+		return -ENODEV;
+found:
 		ret = bone_capemgr_remove_slot(slot);
 		mutex_unlock(&info->slots_list_mutex);
 
-- 
1.8.1.4

